src/note-components/Homepage/homepage.jsx

```diff
@@ -5,23 +5,24 @@ import { Sidebar } from "../Sidebar/sidebar";
 import { useNote } from "../../Context/note-context";
 import { useAuth } from "../../Context/auth-context";
 import { createNote } from "../../note-API/create-note";
-import { useState } from "react";
 import { useNavigate } from "react-router-dom";
 import axios from "axios";
+import { useFilter } from "../../Context/filter-context";
 
 const Homepage = () => {
-const {noteState, noteDispatch} = useNote();
-const {note} = noteState;
+const {noteState, noteDispatch, notes, setNote, tagItem} = useNote();
+const {note, priority} = noteState;
 const {authState} = useAuth();
 const { token } = authState;
-const [notes, setNote] = useState({ title: "", mainContent: "" });
 const navigate = useNavigate();
+const { filterState, finalFilter} = useFilter();
+
 
 const createNoteFunction = async () => {
 if (token) {
```

Comment:
wrap all the routes that needs auth inside the protected/private route so you don't have to check for token every time you call some method need auth like this 
   ```
<Route element={<PrivateRoute />}>
          <Route path="/" element={<HomePage />} />
          <Route path="/bookmarks" element={<BookmarkedPage />} />
          <Route path="/explore" element={<ExplorePage />} />
          <Route path="/profile/:id" element={<ProfilePage />} />
          <Route path="/messages/" element={<MessagePage />} />
        </Route>
```

Reply:
Okay! Thanks for explaining.